src/Microsoft.DotNet.Interactive.Tests/LanguageKernelTests.cs

```diff
@@ -269,19 +269,136 @@ public async Task it_returns_diagnostics(Language language)
 
             await SubmitCode(kernel, source);
 
-            var error = language switch
-            {
-                Language.FSharp => "input.fsx (1,1)-(1,7) typecheck error The value or constructor 'aaaadd' is not defined.",
-                Language.CSharp => "(1,1): error CS0103: The name 'aaaadd' does not exist in the current context"
-            };
+            var diagnosticRange = new LinePositionSpan(
+                new LinePosition(0, 0),
+                new LinePosition(0, 6));
 
             KernelEvents
                 .Should()
                 .ContainSingle<CommandFailed>()
                 .Which
                 .Message
                 .Should()
-                .Be(error);
+                .Be(errorMessage);
+
+            KernelEvents
+                .Should()
+                .ContainSingle<DiagnosticsProduced>(d => d.Diagnostics.Count > 0)
+                .Which
+                .Diagnostics
+                .Should()
+                .ContainSingle(diag => diag.LinePositionSpan == diagnosticRange && diag.Code == code && diag.Message == diagnosticMessage);
+        }
+
+        [Fact]
+        public async Task powershell_produces_diagnostics_from_parse_errors()
+        {
+            var kernel = CreateKernel(Language.PowerShell);
+
+            await kernel.SubmitCodeAsync("::()");
+
+            var diagnosticRange = new LinePositionSpan(
+                new LinePosition(0, 4),
+                new LinePosition(0, 4));
+
+            KernelEvents
+                .Should()
+                .ContainSingle<DiagnosticsProduced>(d => d.Diagnostics.Count > 0)
+                .Which
+                .Diagnostics
+                .Should()
+                .ContainSingle(d => d.LinePositionSpan == diagnosticRange && d.Code == "ExpectedExpression" && d.Message == "An expression was expected after '('.");
+        }
+
+        [Theory]
+        [InlineData(Language.CSharp, @"
+int SomeFunction(int n)
+{
+    return n switch
+    {
+        0 => 0
+    };
+}
+
+", "The switch expression does not handle all possible values of its input type")]
+        [InlineData(Language.FSharp, @"
+let x n =
+    match n with
+    | 0 -> ()
+", "Incomplete pattern matches on this expression.")]
+        public async Task it_returns_diagnostics_on_command_succeeded(Language language, string code, string diagnosticText)
```

Comment:
This one again is a test on diagnostics produced for compile time warning. no need to assert on command succeeded or failuer